### PR TITLE
CFM-891: Place "My groups" block on the public profile

### PR DESCRIPTION
**Pull request by DavidHernandez**
Issue: https://issuetracker.amplexor.com/jira/browse/CFM-891

### Depends on https://github.com/capacity4dev/capacity4more/pull/1155

Screenshot:
my-groups

### CODE REVIEW

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc [hunk 1]
```diff
@@ -191,3 +191,54 @@ function _c4m_user_profile_block_user_confirm_email() {
...
+
+/**
+ * Block definition for the user_my_profile_groups block.
+ *
+ * @return array
+ *   The block definition.
+ */
+function _c4m_user_profile_block_user_my_profile_groups() {
+  module_load_include('inc', 'c4m_features_homepage', 'plugins/content_types/my_groups/my_groups');
+  $user = menu_get_object('user', 1);
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc]: @DavidHernandez 
why do we use `1`  in here?

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc [hunk 2]
```diff
@@ -191,3 +191,54 @@ function _c4m_user_profile_block_user_confirm_email() {
...
+ * Returns the title for the type of membership of a group.
+ */
+function c4m_user_profile_get_group_type_title($type) {
+  $titles = array(
+    'owner' => t('Groups I own'),
+    'administrator' => t('Groups I manage'),
+    'member' => t("Groups I'm member of"),
+    'pending' => t('Membership requested'),
+  );
+  return $titles[$type];
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc]: Add a line spacing above.

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc [hunk 3]
```diff
@@ -191,3 +191,54 @@ function _c4m_user_profile_block_user_confirm_email() {
...
+    'groups' => $rendered_groups,
+  );
+
+  $block['content'] = $rendered_groups ? theme('c4m_user_profile_my_profile_groups', $variables) : FALSE;
+  return $block;
+}
+
+/**
+ * Returns the title for the type of membership of a group.
+ */
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc]: Add docs for the `$type` argument...

Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc]: also the return value

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc [hunk 4]
```diff
@@ -191,3 +191,54 @@ function _c4m_user_profile_block_user_confirm_email() {
...
+      $node_view = node_view($group, 'block_list');
+      $rendered_groups .= drupal_render($node_view);
+    }
+  }
+
+  $variables = array(
+    'groups' => $rendered_groups,
+  );
+
+  $block['content'] = $rendered_groups ? theme('c4m_user_profile_my_profile_groups', $variables) : FALSE;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc]: @DavidHernandez 
It seems that at this point `$rendered_groups` will always returns TRUE ...

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc [hunk 5]
```diff
@@ -191,3 +191,54 @@ function _c4m_user_profile_block_user_confirm_email() {
...
+  return $block;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc]: Add line spacing above

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc [hunk 6]
```diff
@@ -191,3 +191,54 @@ function _c4m_user_profile_block_user_confirm_email() {
 
   return $block;
 }
+
+/**
+ * Block definition for the user_my_profile_groups block.
+ *
+ * @return array
+ *   The block definition.
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc]: `The rendered "my groups" block.`

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc [hunk 7]
```diff
@@ -191,3 +191,54 @@ function _c4m_user_profile_block_user_confirm_email() {
...
+ */
+function _c4m_user_profile_block_user_my_profile_groups() {
+  module_load_include('inc', 'c4m_features_homepage', 'plugins/content_types/my_groups/my_groups');
+  $user = menu_get_object('user', 1);
+  $groups = c4m_features_homepage_get_user_groups_grouped_by_membership($user, array('group'));
+
+  $block = array();
+  $block['subject'] = '';
+  $block['content'] = '';
+  if (is_null($groups)) {
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc]: @DavidHernandez 
It seems that `$groups` can't be null but an empty array.

File: project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc [hunk 8]
```diff
@@ -106,6 +106,7 @@ function c4m_features_homepage_get_user_groups($account = NULL, $allowed_types =
 function c4m_features_homepage_get_user_groups_grouped_by_membership($account = NULL, $allowed_types = array('group')) {
   $groups = c4m_features_homepage_get_user_groups($account, $allowed_types);
 
+  $grouped_groups = array();
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/features/c4m_features_homepage/plugins/content_types/my_groups/my_groups.inc]: @DavidHernandez 
probably should be on a previous PR

**Comment by ordavidil** [comment]
@DavidHernandez 
2 more screenshots please,  thanks ...

**Comment by DavidHernandez** [comment]
my-groups-mobile
my-groups-tablet

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc [hunk 9]
```diff
@@ -200,8 +200,8 @@ function _c4m_user_profile_block_user_confirm_email() {
  */
 function _c4m_user_profile_block_user_my_profile_groups() {
   module_load_include('inc', 'c4m_features_homepage', 'plugins/content_types/my_groups/my_groups');
-  $user = menu_get_object('user');
-  $groups = c4m_features_homepage_get_user_groups_grouped_by_membership($user, array('group'));
+  $account = menu_get_object('user');
+  $groups = c4m_features_homepage_get_user_groups_grouped_by_membership($account, array('group'));
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc]: @DavidHernandez 
why not just changing this one in here:
https://github.com/capacity4dev/capacity4more/pull/1155/files#diff-3f72bf9fc8c0ee2206d46d044126106fR204
?

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc [hunk 10]
```diff
@@ -226,18 +226,18 @@ function _c4m_user_profile_block_user_my_profile_groups() {
     'groups' => $rendered_groups,
   );
 
-  $block['content'] = empty($rendered_groups) ? theme('c4m_user_profile_my_profile_groups', $variables) : FALSE;
+  $block['content'] = !empty($rendered_groups) ? theme('c4m_user_profile_my_profile_groups', $variables) : FALSE;
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc]: please change this one in here:
https://github.com/capacity4dev/capacity4more/pull/1155/files#diff-3f72bf9fc8c0ee2206d46d044126106fR229

File: project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc [hunk 11]
```diff
@@ -200,8 +200,8 @@ function _c4m_user_profile_block_user_confirm_email() {
  */
 function _c4m_user_profile_block_user_my_profile_groups() {
   module_load_include('inc', 'c4m_features_homepage', 'plugins/content_types/my_groups/my_groups');
-  $user = menu_get_object('user');
-  $groups = c4m_features_homepage_get_user_groups_grouped_by_membership($user, array('group'));
+  $account = menu_get_object('user');
```
Comment (ordavidil) [inline project/profiles/capacity4more/modules/c4m/user/c4m_user_profile/c4m_user_profile.block.inc]: please change this one in here:
https://github.com/capacity4dev/capacity4more/pull/1155/files#diff-3f72bf9fc8c0ee2206d46d044126106fR203

File: project/profiles/capacity4more/themes/c4m/kapablo/theme/preprocess.inc [hunk 12]
```diff
@@ -616,8 +616,8 @@ function kapablo_preprocess_html(&$variables) {
  * Implements template_preprocess_user_profile().
  */
 function kapablo_preprocess_user_profile(&$variables) {
-  $blocks = array(block_load('c4m_user_profile', 'user_my_profile_groups'));
-  $renderable_blocks = _block_render_blocks($blocks);
+  $block = block_load('c4m_user_profile', 'user_my_profile_groups');
+  $renderable_blocks = _block_render_blocks(array($block));
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/theme/preprocess.inc]: change it in here:
https://github.com/capacity4dev/capacity4more/pull/1155/files#diff-dcea7c519f8e4ed9be3a183c61bc4ff6R620

File: project/profiles/capacity4more/themes/c4m/kapablo/theme/preprocess.inc [hunk 13]
```diff
@@ -616,8 +616,8 @@ function kapablo_preprocess_html(&$variables) {
  * Implements template_preprocess_user_profile().
  */
 function kapablo_preprocess_user_profile(&$variables) {
-  $blocks = array(block_load('c4m_user_profile', 'user_my_profile_groups'));
-  $renderable_blocks = _block_render_blocks($blocks);
+  $block = block_load('c4m_user_profile', 'user_my_profile_groups');
```
Comment (ordavidil) [inline project/profiles/capacity4more/themes/c4m/kapablo/theme/preprocess.inc]: change it in here:
https://github.com/capacity4dev/capacity4more/pull/1155/files#diff-dcea7c519f8e4ed9be3a183c61bc4ff6R619

**Comment by ordavidil** [comment]
@DavidHernandez 
__AFTER__ fixing the comments here __under #1155__ please re-roll the current PR with `feature/CFM-448-my-groups-block-for-profile`

**Pull request merged**